config/base/src/runtime_upgrades.rs

```diff
@@ -208,7 +196,7 @@ pub mod handle {
 
     impl<T: Send + Sync + Debug> Reload<T> for Singleton<T> {
         fn reload(&self, item: T) -> Result<()> {
-            match &mut *self.inner.lock()? {
+            match &mut *self.inner.lock().expect("Valid") {
```

Comment:
then why don't we use non poisoning mutex as in sumeragi?

Reply:
We should. It's like this, because this PR was started before we had `parking_lot::Mutex`